lib/index.js

```diff
@@ -236,16 +267,130 @@ module.exports = class OtpRunner {
+
+    // add tasks to download files to the base folder
+    if (
+      this.manifest.baseFolderDownloads &&
+        this.manifest.baseFolderDownloads.length > 0
+    ) {
+      this.manifest.baseFolderDownloads.forEach(url => {
+        const splitUrl = url.split('/')
+        downloadTasks.push({
+          dest: path.join(
+            this.getTargetFolder(),
+            splitUrl[splitUrl.length - 1]
+          ),
+          url
+        })
+      })
+    }
+
+    // add tasks to download various GTFS files and then rename them according
+    // to OTP 2.x file naming requirements
+    if (
+      this.manifest.gtfsDownloads &&
+        this.manifest.gtfsDownloads.length > 0
+    ) {
+      this.manifest.gtfsDownloads.forEach(url => {
+        const splitUrl = url.split('/')
+        downloadTasks.push({
+          dest: path.join(
+            this.getTargetFolder(),
+            `gtfs-${splitUrl[splitUrl.length - 1]}`
+          ),
+          url
+        })
+      })
+    }
+
+    if (!this.manifest.buildGraph && this.manifest.runServer) {
+      // Manifest says to run the server without building a graph.
+      // Therefore, download a Graph.obj file.
+      downloadTasks.push({
+        dest: path.join(this.getTargetFolder(), this.getGraphObjName()),
+        url: this.manifest.graphObjUrl
+      })
+    }
+
+    // download files asynchronously
+    this.status.totalFilesToDownload = downloadTasks.length
+    await this.updateStatus(
+      `Downloading ${this.status.totalFilesToDownload} files...`,
+      10
+    )
+    await Promise.all(
+      downloadTasks.map(download => this.makeDownloadTask(download))
+    )
+  }
+
+  /**
+   * Make OTP build arguments according to the specified OTP version
+   */
+  makeBuildArgs () {
+    const args = getBaseOTPArgs(this.manifest.jarFile)
+    args.push('--build')
+    if (this.isUsingOtp2x()) {
+      args.push('--save')
+    }
+    args.push(this.getTargetFolder())
+    return args
+  }
+
+  /**
+   * Builds a graph using OTP if needed.
    */
   async buildGraph () {
+    // build graph if needed
+    if (!this.manifest.buildGraph) return
+
+    await this.updateStatus(
+      'Building graph',
+      this.manifest.runServer ? 30 : 50
+    )
+
+    // write build-config.json file if contents are supplied in this.manifest
+    if (this.manifest.buildConfigJSON) {
+      await fs.writeFile(
+        path.join(this.getTargetFolder(), 'build-config.json'),
+        this.manifest.buildConfigJSON
+      )
+    }
+
     // in build mode, use execa which simplifies a few things
     // The execa options argument creates a combined stdout and stderr output
     // stream via setting the `all` flag to true.
-    const args = getBaseOTPArgs(this.manifest.jarFile)
-    args.push('--build')
-    args.push(path.join(this.manifest.graphsFolder, this.manifest.routerName))
-    const subprocess = execa('java', args, { all: true })
+    const subprocess = execa('java', this.makeBuildArgs(), { all: true })
```

Comment:
I think it would be helpful to log the java command anytime we execute one.

Reply:
The serve command is already logged, so I added a log for this command in 2610ee9df6cb1878f067a5e2c1c277eed33836fe